src/test/java/stepDefinition/HomePageStepDef.java

```diff
@@ -1,24 +1,24 @@
 package stepDefinition;
 
-import static org.hamcrest.CoreMatchers.is;
-import static org.junit.Assert.assertThat;
-
 import cucumber.api.java.en.Given;
+import cucumber.api.java.en.Then;
+import pages.HomePage;
 
 import static helpers.Reflection.getPage;
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
 
 public class HomePageStepDef extends AbstractStepDef {
 
-    @Given("^Home page is displayed$")
-    public void homePageIsDisplayed() {
+    @Given("^user is on Home page$")
+    public void userIsOnHomePage() {
         webDriver.navigate().to(fileReader.getApplicationUrl());
-        assertThat("Page is displayed", webDriver.getCurrentUrl(), is(getPage("HomePage").getPageUrl()));
+        assertThat("Page is displayed", webDriver.getCurrentUrl(), is(getPage(HomePage.class.getSimpleName()).getPageUrl()));
     }
 
-    @Given("^'(.*)' page is displayed$")
-    public void homePageIsDisplayed(String pageName) {
+    @Then("^'(.*)' page is displayed$")
+    public void pageIsDisplayed(String pageName) {
         assertThat(String.format("%s page is displayed", pageName), webDriver.getCurrentUrl(), is(getPage(pageName.concat("Page")).getPageUrl()));
```

Comment:
this one: 'String.format("%s page is displayed", pageName)' could be assigned to a Sting variable and then use this variable in assert. In such a way you obtain a shorter and more readable code line.

Reply:
And second, pay attention to the assert message text, message will be displayed in case when assert fail. So the current text meaning is wrong. It should sound something like "%s page **isn't** (cannot be) displayed". And after I took a look on other messages I've found that there are other ones that contain the same wrong formulated text. So find them all and change.

Reply:
disagree:
1. there's no sens to same the string from assert in a String variable as you need it in just one place, it's just wasting of memory, instead you can break assert in more lines if you want to be more readable 
2. using assertThat it will show what what you expected and the actual value, but message is just a title/header and you expect that you will have a positive case not a negative 
3. as we want our message to be seen not just on fail, we can add custom asserts that will have a **log.info("message parsed")**